src/RevSysCalls.cc

```diff
@@ -620,6 +620,50 @@ EcallStatus RevCore::ECALL_openat() {
   return EcallLoadAndParseString( pathname, action );
 }
 
+// 1024, int rev_open(const char *filename, int flags)
+EcallStatus RevCore::ECALL_open() {
+  auto& EcallState = Harts.at( HartToExecID )->GetEcallState();
+  if( EcallState.bytesRead == 0 ) {
+    output->verbose(
+      CALL_INFO, 2, 0, "ECALL: openat called by thread %" PRIu32 " on hart %" PRIu32 "\n", ActiveThreadID, HartToExecID
+    );
+  }
+  auto pathname = RegFile->GetX<uint64_t>( RevReg::a0 );
+
+  auto flags = RegFile->GetX<int>(RevReg::a1);
+
+  /* Read the filename from memory one character at a time until we find '\0' */
+
+  auto action   = [&] {
+    // Do the open on the host
+    std::string const full_path =
+      std::filesystem::current_path().append(EcallState.string).string();
+    auto fd  = open( full_path.c_str(), flags);
+
+    if(fd == -1) {
+       output->fatal(
+          CALL_INFO,
+          -1,
+          "Core %" PRIu32 "; Hart %" PRIu32 "; Thread %" PRIu32 " tried to close file descriptor %" PRIu32
+          " but did not have access to it\n",
+          id,
+          HartToExecID,
+          ActiveThreadID,
+          fd
+       );
```

Comment:
I would not make it a fatal error message, but simply not call `AddFD()` if `fd == -1`:
```c++
if (fd != -1) Harts.at( HartToExecID )->Thread->AddFD( fd );
```
It will return `-1` to the caller by setting `a0` to `fd`, error or not, before returning.

It is up to the caller what to do in case of error.

Also, your error message says "tried to **close** file descriptor 4294967295 but did not have access to it". The file descriptor should not be printed if it's `-1`, should not be printed as unsigned, and the reason for the failure varies (it is in `errno`).